File: Sanchime.Identity/Entities/Role.cs
namespace Sanchime.Identity.Entities;

public class Role : IdentityEntity
{
    public required string Code { get; set; }

    public required string Name { get; set; }

    public string? Description { get; set; }

    public long? ParentId { get; set; }

    public Role? Parent { get; set; }

    public List<User> Users { get; set; } = [];

    public ICollection<Role> Children { get; } = [];

    public IList<Permission> Permissions { get; set; } = [];

    public IList<Menu> Menus { get; set; } = [];

    public IList<RoleMenu> RoleMenus { get; set; } = [];
}

File: Frontends/sanchime-identity-vue-app/src/models/role.ts
export interface RoleResponse {
    id: number
    code: string
    name: string
    description: string | null
    isEnabled: boolean
}

export interface AddRoleRequest {
    code: string
    name: string
    description: string | null
}

export interface UpdateRoleRequest {
    code: string
    name: string
    description: string | null
    isEnabled: boolean
}

export interface UpdateRolePermissionRequest {
    permissions: number[]
}